File: test 3/3/stack.cpp
#include "stack.h"

struct StackElement
{
    char value;
    StackElement *next;
};

struct Stack
{
    StackElement *head;
};

bool isEmpty(Stack *stack)
{
    return (stack->head == nullptr);
}

Stack *createStack()
{
    Stack *newStack = new Stack;
    newStack->head = nullptr;
    return newStack;
}

void push(Stack *stack, char value)
{
    StackElement *newStack = new StackElement;
    newStack->value = value;
    newStack->next = stack->head;
    stack->head = newStack;
}

char pop(Stack *stack)
{
   if (isEmpty(stack))
   {
       return -17;
   }
   StackElement *temp = nullptr;
   temp = stack->head;
   char value = temp->value;

   stack->head = stack->head->next;
   delete temp;
   temp = nullptr;
   return value;
}

void emptyMemory(Stack *stack)
{
    while (!isEmpty(stack))
    {
        pop(stack);
    }
    delete stack;
}

char peek(Stack *stack)
{
    if (isEmpty(stack))
        return 'k';
    return stack->head->value;
}

File: sem 2/1 hw/1/src/main/java/group144/shervashidze/Stack.java
package group144.shervashidze;

public class Stack {
    private Element head = null;
    private int length = 0;

    public boolean isEmpty() {
        return length == 0;
    }

    public void add(int value) {
        head = new Element(value, head);
        length += 1;
    }

    public int pop() {
        if (isEmpty()){
            return 42;
        }
        int result = head.value;
        head = head.next;
        length -= 1;
        return result;
    }

    public int peek() {
        if (isEmpty()){
            return 42;
        }
        return head.value;
    }

    public int getLength() {
        return length;
    }

    private class Element {
        private int value;
        private Element next;

        Element(int value, Element next) {
            this.value = value;
            this.next = next;
        }
    }
}